File: client/app/src/main/java/ru/spbstu/gyboml/clientcore/GameListener.java
package main.java.ru.spbstu.gyboml.clientcore;

import com.esotericsoftware.kryonet.Connection;
import com.esotericsoftware.kryonet.Listener;

import main.java.ru.spbstu.gyboml.GybomlClient;
import ru.spbstu.gyboml.core.PlayerType;
import ru.spbstu.gyboml.core.net.GameResponses;

public class GameListener extends Listener {

    private GybomlGame game;

    GameListener(GybomlGame game) { this.game = game; }

    @Override
    public void received(Connection connection, Object object) {
        if (object instanceof GameResponses.Shooted) { shoted(connection, (GameResponses.Shooted)object); }
    }

    private void shoted(Connection connection, GameResponses.Shooted object) {
        PlayerType whoShoted = object.yourShoot ? GybomlClient.getPlayerType() : GybomlClient.getPlayerType().reverted();
        game.physicalScene.generateShot(GybomlClient.getPlayerType(), game.shotType);
        game.soundEffects.shot.play(1.f);
    }
}

File: client/app/src/ru/spbstu/gyboml/multiplayer/GameListener.kt
package ru.spbstu.gyboml.multiplayer

import com.badlogic.gdx.Gdx
import com.esotericsoftware.kryonet.Connection
import com.esotericsoftware.kryonet.Listener
import ru.spbstu.gyboml.core.net.GameMessage.*
import ru.spbstu.gyboml.core.net.GameResponses.*

class GameListener(private val game: Game) : Listener() {
    override fun received(connection: Connection?, message: Any?) = when (message) {
        is GameExited -> { game.dispose(); Gdx.app.exit() }

        is CreateBlock -> game.physicalScene.createBlock(message)
        is UpdateBlock -> game.physicalScene.updateBlock(message)
        is RemoveBlock -> game.physicalScene.removeBlock(message)

        is CreateShot -> game.physicalScene.createShot(message)
        is UpdateShot -> game.physicalScene.updateShot(message)
        is RemoveShot -> game.physicalScene.removeShot()

        else -> {}
    }
}